caps/import_utils.py

```diff
@@ -69,59 +69,39 @@ def add_gss_codes(filename):
         if not pd.isnull(authority_code):
-            authority_match = authority_df[authority_df['local-authority-code'] == authority_code]
-            plans_df.at[index, 'gss_code'] = authority_match['gss-code'].values[0]
+            authority_match = authority_df[authority_df['local-authority-code']
+                                           == authority_code]
+            plans_df.at[index,
+                        'gss_code'] = authority_match['gss-code'].values[0]
 
     plans_df.to_csv(open(filename, "w"), index=False, header=True)
 
+
 def add_extra_authority_info(filename):
 
     authority_df = pd.read_csv(AUTHORITY_DATA)
     plans_df = pd.read_csv(filename)
 
-    rows = len(plans_df['council'])
-    plans_df['authority_type'] = pd.Series([None] * rows, index=plans_df.index)
-    plans_df['wdtk_id'] = pd.Series([None] * rows, index=plans_df.index)
-    plans_df['mapit_area_code'] = pd.Series([None] * rows, index=plans_df.index)
-    plans_df['country'] = pd.Series([None] * rows, index=plans_df.index)
-    plans_df['gss_code'] = pd.Series([None] * rows, index=plans_df.index)
-
-    for index, row in plans_df.iterrows():
-        authority_code = row['authority_code']
-        if not pd.isnull(authority_code):
-            authority_match = authority_df[authority_df['local-authority-code'] == authority_code]
-            country = None
-            country = authority_match['nation'].values[0]
-            plans_df.at[index, 'authority_type'] = authority_match['local-authority-type'].values[0]
-            plans_df.at[index, 'wdtk_id'] = authority_match['wdtk_ids'].values[0]
-            plans_df.at[index, 'mapit_area_code'] = authority_match['mapit_area_code'].values[0]
-            plans_df.at[index, 'country'] = country
-            plans_df.at[index, 'gss_code'] = authority_match['gss-code'].values[0]
-
-            # All authorities in Wales, Scotland and Northern Ireland are unitary
-            if country in ['Wales', 'Scotland', 'Northern Ireland']:
-                plans_df.at[index, 'authority_type'] = 'UA'
-    plans_df.to_csv(open(filename, "w"), index=False, header=True)
+    df = authority_df[["local-authority-code", "local-authority-type",
+                       "wdtk-id", "mapit-area-code", "nation", "gss-code"]]
 
-def add_combined_authority_gss_codes(filename):
-    plans_df = pd.read_csv(filename)
-    combined_authority_df = pd.read_csv(COMBINED_AUTHORITY_DATA)
+    df = df.rename(columns={"local-authority-code": "authority_code",
+                            "local-authority-type": "authority_type",
+                            "wdtk-id": "wdtk_id",
+                            "mapit-area-code": "mapit_area_code",
+                            "nation": "country",
+                            "gss-code": "gss_code"
+                            })
 
-    aliases = {'newcastle upon tyne, north tyneside and northumberland combined authority': 'north of tyne'}
+    df = plans_df.merge(df, how="left")
```

Comment:
A comment to indicate the basis of this merge would be helpful. I assume it's the gss_code? Or does it just need to have one of the columns to match?

Reply:
Sure, it's the intersection of the columns of the two dataframes (so in this case 'authority_code'). I'll change this so the merge on is explicit.

Reply:
Addressing this also led to a change to explicitly drop the columns that are going to be merged in (as the csv may have been updated previously, and may already contain them).